lib/chef/knife/azurerm_base.rb

```diff
@@ -94,7 +94,7 @@ def authentication_details
         else
           token_details = token_details_for_linux()
         end
-        check_token_validity(token_details)
+        token_details = check_token_validity(token_details)
```

Comment:
Currently `check_token_validity` method is checking the token validity, refreshing the token by calling `azure vm image list-skus westus RedHat RHEL` command and returning the `token_details`. The name of the method doesn't suggest that.